elliottlib/imagecfg.py

```diff
@@ -22,32 +24,26 @@ def base_only(self):
         """
         return self.config.base_only
 
-    def get_latest_build_info(self, product_version):
+    def get_latest_build_info(self, product_version_dict, tag_set, latest_builds, brew_session):
```

Comment:
As talked before, I think the signature of method is weird because it does more than it sounds like, and arguments for this function is hard to understand. I think this need to changed to something like:

1. move this function to `brew.py` rather than a member function of `ImageMetadata` because it query infos for multiple Brew builds and not for a specific image.

2. Change the signature of this function like
``` def
def get_latest_builds(tag, component_names)
```
3. Then you can either use `brew_session.getLatestBuilds` to get all builds with specified tag then filter the result with the list/set in `component_names`, or use brew multi-call to query for specified components like:
``` python
 with brew_session.multicall(strict=True):
        for name in component_names:
            yield brew_session.getLatestBuilds(tag, package=name)
```

Reply:
the tag needs to be connected with `{branch}-candidate`, also need to pass product_version_dict, also I haven't checked previous value of `self.name` if it's sth can't fetch from somewhere else we need to at least have this to pass in:
```
def get_latest_builds(tag, component_names, branch, name)
```
and I can put `latest_builds` as a global variable into `brew.py` and put  `brew_session` into `get_latest_builds`